src/mono/wasm/debugger/BrowserDebugProxy/EvaluateExpression.cs

```diff
@@ -27,38 +27,54 @@ private class FindVariableNMethodCall : CSharpSyntaxWalker
             public List<InvocationExpressionSyntax> methodCall = new List<InvocationExpressionSyntax>();
             public List<MemberAccessExpressionSyntax> memberAccesses = new List<MemberAccessExpressionSyntax>();
             public List<object> argValues = new List<object>();
+            public Dictionary<string, JObject> memberAccessValues = new Dictionary<string, JObject>();
```

Comment:
Does this need to be synchronized across multiple threads ?
Does this need to be flushed when debugged methods have side-effects ?
Does this need to be flushed when hot-reload ?

Reply:
We instantiate an object of type `FindVariableNMethodCall` for each expression. We don't need to care about flush or synchronize across multiple threads.